### DISCUSSION TITLE
Vim keybinding mode

### DISCUSSION THREAD

**Discussion by thedeveloperr** [outline/rich-markdown-editor#331] [Ideas] [answered]
Is there any plan for vim keybindings ? Also in case no plan for this, any tips for adding vim keybinding so i can try to add it ?

**Comment by tommoor** [comment, marked as answer]
Hi, you should provide this functionality as an extension – we could potentially include it in the main repo as an optional add-on once complete if you're interested. In terms of how to achieve this I'd follow the pattern set out here:

https://github.com/outline/rich-markdown-editor/blob/develop/src/plugins/Keys.ts

In the context of the extension you have `this.editor.commands` which should contain everything you need. You can then pass the resulting extension to the `extensions` prop on the editor.
